src/transformers/audio_utils.py

```diff
@@ -581,6 +581,214 @@ def spectrogram(
+        power (`float`, optional, default=1.0):
+            Determines the type of spectrogram: 1.0 for amplitude, 2.0 for power, None for complex.
+        center (`bool`, optional, default=True):
+            Whether to center-pad the waveform frames.
+        pad_mode (`str`, optional, default="reflect"):
+            The padding strategy when `center` is True.
+        onesided (`bool`, optional, default=True):
+            If True, returns a one-sided spectrogram for real input signals.
+        preemphasis (`float`, optional):
+            Applies a pre-emphasis filter to each frame.
+        mel_filters (`np.ndarray`, optional):
+            Mel filter bank for converting to mel spectrogram.
+        mel_floor (`float`, optional, default=1e-10):
+            Floor value for mel spectrogram to avoid log(0).
+        log_mel (`str`, optional):
+            Specifies log scaling strategy; options are None, "log", "log10", "dB".
+        reference (`float`, optional, default=1.0):
+            Reference value for dB conversion in log_mel.
+        min_value (`float`, optional, default=1e-10):
+            Minimum floor value for log scale conversions.
+        db_range (`float`, optional):
+            Dynamic range for dB scale spectrograms.
+        remove_dc_offset (`bool`, optional):
+            Whether to remove the DC offset from each frame.
+        dtype (`np.dtype`, optional, default=np.float32):
+            Data type of the output spectrogram.
+
+    Returns:
+        List[`np.ndarray`]: A list of spectrogram arrays, one for each input waveform.
+    """
+    window_length = len(window)
+
+    if fft_length is None:
+        fft_length = frame_length
+
+    if frame_length > fft_length:
+        raise ValueError(f"frame_length ({frame_length}) may not be larger than fft_length ({fft_length})")
+
+    if window_length != frame_length:
+        raise ValueError(f"Length of the window ({window_length}) must equal frame_length ({frame_length})")
+
+    if hop_length <= 0:
+        raise ValueError("hop_length must be greater than zero")
+
+    # Check the dimensions of the waveform
+    for waveform in waveform_list:
+        if waveform.ndim != 1:
+            raise ValueError(f"Input waveform must have only one dimension, shape is {waveform.shape}")
+
+    # Check if waveform is complex
+    for waveform in waveform_list:
+        if np.iscomplexobj(waveform):
+            raise ValueError("Complex-valued input waveforms are not currently supported")
+
+    # Center pad the waveform
+    if center:
+        padding = [(int(frame_length // 2), int(frame_length // 2))]
+        waveform_list = [
+            np.pad(
+                waveform,
+                padding,
+                mode=pad_mode,
+            )
+            for waveform in waveform_list
+        ]
+    original_waveform_lengths = [
+        len(waveform) for waveform in waveform_list
+    ]  # these lengths will be used to remove padding later
```

Comment:
nit - library convention is comments on the line above to avoid line splitting like this 

```suggestion
    # these lengths will be used to remove padding later
    original_waveform_lengths = [len(waveform) for waveform in waveform_list]
```